src/vm/types.rs

```diff
@@ -384,6 +390,30 @@ impl fmt::Display for Value {
 }
 
 impl PrincipalData {
+    pub fn parse_qualified_contract_principal(literal: &str) -> Result<PrincipalData> {
+        let split: Vec<_> = literal.splitn(2, ".").collect();
+        if split.len() != 2 {
+            return Err(RuntimeErrorType::ParseError(
+                "Invalid principal literal: expected a `.` in a qualified contract name".to_string()).into());
+        }
+        let sender = Self::parse_standard_principal(split[0])?;
+        let name = split[1].to_string();
```

Comment:
Do we want to enforce a name length here?

Reply:
I don't think so -- if we are going to enforce a name length, I'd prefer to minimize the number of places such an enforcement is defined. The eventual type for qualified contract names will implement that enforcement.